membership-attribute-service/app/controllers/SalesforceHookController.scala

```diff
@@ -118,33 +118,27 @@ class SalesforceHookController extends LazyLogging {
         // Zuora is the master for product info, so we use the tier from Zuora regardless of what Salesforce sends
         val tierFromZuora = membershipSubscription.plan.charges.benefit.id
 
-        // If we have the card expiry date in Stripe, add them to Dynamo too.
+        // If we have the card expiry date in Stripe, add them to Dynamo too inside a Wallet construct.
         // TODO - refactor to use touchpoint.paymentService - requires membership-common model tweak first.
-        val cardExpiryFromStripeF = (for {
+        val walletF = for {
           account <- OptionT(touchpoint.zuoraService.getAccount(membershipSubscription.accountId).map(Option(_)))
           paymentMethodId <- OptionT(Future.successful(account.defaultPaymentMethodId))
           paymentMethod <- OptionT(touchpoint.zuoraService.getPaymentMethod(paymentMethodId).map(Option(_)))
           customerToken <- OptionT(Future.successful(paymentMethod.secondTokenId))
           stripeCustomer <- OptionT(touchpoint.stripeService.Customer.read(customerToken).map(Option(_)))
         } yield {
-          (stripeCustomer.card.exp_month, stripeCustomer.card.exp_year)
-        }).run
+          Wallet(membershipCard = Some(CardDetails.fromStripeCard(stripeCustomer.card, Membership.id)))
```

Comment:
forProduct should really be a Product for as long as possible i.e. `Membership` rather than `Membership.id`